libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/URLConnectionNetworkActions.java

```diff
@@ -79,6 +81,28 @@ private static void withJdkHttpConnection(CheckedConsumer<HttpURLConnection, Exc
         }
     }
 
+    private static void withJdkHttpsConnection(CheckedConsumer<HttpsURLConnection, Exception> connectionConsumer) throws Exception {
+        var conn = EntitledActions.createHttpsURLConnection();
+        // Be sure we got the connection implementation we want
+        assert HttpsURLConnection.class.isAssignableFrom(conn.getClass());
+        try {
+            connectionConsumer.accept((HttpsURLConnection) conn);
+        } catch (java.net.ConnectException e) {
+            // It's OK, it means we passed entitlement checks, and we tried to connect
+        }
+    }
+
+    private static void withJdkFtpConnection(CheckedConsumer<URLConnection, Exception> connectionConsumer) throws Exception {
+        var conn = EntitledActions.createFtpURLConnection();
```

Comment:
Should we be making an effort to close these connections?

I don't actually see a `close()` method or similar. Maybe there's nothing that needs to be closed.

Reply:
Yeah, that seemed suspicious to me as well but there is no close() method (or similar).